### PR TITLE
add repository and api for fcm #181

### LINKED ISSUE #181: Create API and Reposiotry for FCM, add implementation of sending server
## Kind (Required)

- Improvement

 ## Overview (Required)
disassemble this issue ☞ https://github.com/DroidKaigi/conference-app-2021/issues/41

- crate api and repository
- inject repository to messagingService

 ## Links	

- https://droidkaigi.stoplight.io/docs/ssot-api/openapi/reference/external.v1.yaml/paths/~1devices~1%7BdeviceId%7D/put

### PR DESCRIPTION
**Pull request by RyuNen344**
## Issue
- close #181

## Overview (Required)
- api of creation deviceId requestbody was not defined in openapi, but I thought need it(is it correct?)
- moved FeedRepository to model.feeder.repository package(I uesd app of 2020 as a reference)
- renamed MessagingServicie ☞ https://github.com/DroidKaigi/conference-app-2021/pull/170#discussion_r584284113

## Links
-

## Screenshot
Before | After
:--: | :--:
 |

### CODE REVIEW

**Comment by mhidaka** [comment]
👀 
@RyuNen344 Awesome! the review of this p-r will be in the today morning, please wait a minute.

**Comment by RyuNen344** [comment]
thx! ok! 🙇

**Comment by takahirom** [comment]
👀

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/KtorDeviceApi.kt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import io.ktor.http.ContentType
+import io.ktor.http.contentType
+
+open class KtorDeviceApi(
+    private val authApi: AuthApi,
+    private val httpClient: HttpClient,
+) : DeviceApi {
+
+    override suspend fun create(): DeviceInfo =
+        authApi.authenticated {
```
Comment (takahirom) [inline data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/KtorDeviceApi.kt:20]: Thanks for reading the other code and incorporating the code you need! 💯

File: android/src/main/java/io/github/droidkaigi/feeder/notification/AppMessagingService.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@AndroidEntryPoint
+class AppMessagingService : FirebaseMessagingService() {
+
+    @Inject
+    lateinit var deviceRepository: DeviceRepository
+
+    private val serviceJob: Job = Job()
```
Comment (takahirom) [inline android/src/main/java/io/github/droidkaigi/feeder/notification/AppMessagingService.kt:20]: Referencing implementations such as MainScope() and viewModelScope, SupervisorJob() may be better.
This does not need to be implemented in this PR.

File: data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/request/DevicePostRequest.kt [hunk 3]
```diff
@@ -0,0 +1,8 @@
+package io.github.droidkaigi.feeder.data.request
+
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class DevicePostRequest(val osName: String)
+
+expect fun platform(): String
```
Comment (takahirom) [inline data/api/src/commonMain/kotlin/io/github/droidkaigi/feeder/data/request/DevicePostRequest.kt:8]: 🆒

**Review by takahirom — APPROVED** [review]
Thanks for the best readable implementation!

**Pull request merged**